lib/dyph3/two_way_differs/heckel_diff.rb

```diff
@@ -1,85 +1,178 @@
 module Dyph3
   module TwoWayDiffers
+
     class HeckelDiff
+      # Algorithm adapted from http://www.rad.upenn.edu/sbia/software/basis/apidoc/v1.2/diff3_8py_source.html
+
+      def self.execute_diff(old_text_array, new_text_array)
+        raise ArgumentError, "Argument is not an array." unless old_text_array.is_a?(Array) && new_text_array.is_a?(Array)
+
+        diff_result = diff(old_text_array, new_text_array)
+
+        # convert to typed differ's output (wrapped with change types eg. Add, Delete, Change)
+        convert_to_typed_ouput(diff_result, old_text_array, new_text_array)
+      end
+
       # Two-way diff based on the algorithm by P. Heckel.
-      # @param [in] text_a Array of lines of first text.
-      # @param [in] text_b Array of lines of second text.
-      # @returns Array of arrays [cmd, our_lo, our_hi, their_lo, their_hi]
-      def self.diff(text_a, text_b)
-        uniq = find_uniq_matches(text_a, text_b)
-        # start with a1, b1 being the lines before the first change.
-        # for each pair of lines in uniq which definitely match eachother:
-        a1, b1 = [0, 0]
-        d = []
-
-        uniq.each do |a_uniq, b_uniq|
-          lo_a, lo_b = move_to_next_difference(text_a, text_b, a1, b1)
-          a1, b1 = [a_uniq + 1, b_uniq + 1]
-
-          # (a_uniq < a1 || b_uniq < b1) == true guarentees there is not a change (since we walked a1 and b1 to changes before this section, and at the end of each block)
-          # a1 and b1 are always the lines right before the next change.
-          next if a_uniq < lo_a || b_uniq < lo_b
-
-          # we know a_uniq to be the next line which has a corresponding b_uniq. so a1 = last line of potential change (as does b1)
-          hi_a, hi_b = move_to_prev_difference(text_a, text_b, lo_a, lo_b, a_uniq - 1, b_uniq - 1)
-          d << Dyph3::Support::AssignAction.get_action(lo_a: lo_a, lo_b: lo_b, hi_a: hi_a, hi_b: hi_b)
-        end
+      # @param [in] left Array of anything implementing hash and equals
+      # @param [in] right Array of anything implementing hash and equals
+      def self.diff(left, right)
+        differ = HeckelDiff.new(left,right)
+        differ.perform_diff
+      end
```

Comment:
I think it would be preferable to elide the static methods, and prefer just the object style API.